### PR TITLE
Remove Long Deprecated SmartSim Modules

### PR DESCRIPTION
**Pull request by amandarichardsonn**

### CODE REVIEW

**Review by al-rigazzi — APPROVED** [review]
LGTM, pending review by issue reporter!

**Comment by amandarichardsonn** [comment]
@MattToast there is a link error with the SmartSim documentation changelog update under Detailed Notes, that correction is in the read-yaml PR so ignore!

**Review by MattToast — APPROVED** [review]
LGTM! Thanks for cleaning up the SmartSim repo!

**Pull request merged**
